swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/LockingService.kt

```diff
@@ -18,12 +18,31 @@ package com.netflix.spinnaker.swabbie
 
 import com.netflix.spinnaker.config.LockingConfigurationProperties
 import com.netflix.spinnaker.kork.lock.LockManager
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import java.time.Duration
+import java.util.concurrent.atomic.AtomicInteger
 
 class LockingService(
   private val lockManager: LockManager,
   private val lockingConfigurationProperties: LockingConfigurationProperties
 ) : LockManager by lockManager {
+  private val log: Logger = LoggerFactory.getLogger(javaClass)
+
+  val numLocksCurrentlyHeld = AtomicInteger(0)
+
   val swabbieMaxLockDuration: Duration
     get() = Duration.ofMillis(lockingConfigurationProperties.maximumLockDurationMillis!!)
+
+  val swabbieMaxConcurrentLocks: Int
+    get() = lockingConfigurationProperties.maxConcurrentLocks
+
+  @Synchronized override fun acquireLock(lockOptions: LockManager.LockOptions, onLockAcquiredCallback: Runnable): LockManager.AcquireLockResponse<Void> {
```

Comment:
I wonder if you could take advantage of the fact that there is an onLockAcquiredCallback here and wrap something around that to do a try { increment } finally { decrement }